src/Microsoft.AspNet.Hosting/Internal/HostingEngine.cs

```diff
@@ -79,6 +80,8 @@ public virtual IDisposable Start()
                         contextAccessor.HttpContext = httpContext;
                         await application(httpContext);
                     }
+
+                    return httpContext;
```

Comment:
@halter73 He tried that first, see https://github.com/aspnet/Hosting/pull/355#discussion_r39513731